test/functional/transactions_tests.js

```diff
@@ -121,4 +121,77 @@ describe('Transactions', function() {
       }
     });
   });
+
+  describe('TransientTransactionError', function() {
+    it('should have a TransientTransactionError label inside of a transaction', {
+      metadata: { requires: { topology: 'replicaset', mongodb: '>=4.0.0' } },
+      test: function(done) {
+        const configuration = this.configuration;
+        const client = configuration.newClient({ w: 1 }, { useUnifiedTopology: true });
+
+        client.connect((err, client) => {
+          const session = client.startSession();
+          const db = client.db(configuration.db);
+          const coll = db.collection('transaction_error_test');
+
+          session.startTransaction();
+          coll.insertOne({ a: 1 }, { session }, err => {
+            expect(err).to.not.exist;
+            expect(session.inTransaction()).to.be.true;
+
+            db.executeDbAdminCommand(
+              {
+                configureFailPoint: 'failCommand',
+                mode: { times: 2 },
+                data: { failCommands: ['commitTransaction'], closeConnection: true }
+              },
+              () => {
+                expect(session.inTransaction()).to.be.true;
+                session.commitTransaction(err => {
```

Comment:
We are supposed to be testing for a `TransientTransacitonError`, which AFAIK should not occur on a `commitTransaction` command. I think the right way to test this is to do the following:

1. Start a transaction
2. Insert a document from within the transaction
3. Enable Fail Point
4. Attempt to insert another document from within the transaction
5. Verify that step 4 fails with a MongoNetworkError
6. Verify that step 4's error has the label `TransientTransactionError`

Reply:
It looks like that error _is_ occurring on a `commitTransaction` command since the test is passing.